From a423be24de6f008c7c6201e9cf2ae6b8e70807b5 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
Date: Sat, 10 Jan 2015 03:34:51 -0800
Subject: [PATCH 2002/2012] Update CheckArches to fallback on Intel/ppc if ld
 -v doesn't report supported architectures

Older versions of ld64 (eg: Xcode 3.x) do not report this.

http://www.llvm.org/bugs/show_bug.cgi?id=22182

Signed-off-by: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
---
 make/platform/clang_darwin.mk | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git llvm_release_38/projects/compiler-rt/make/platform/clang_darwin.mk macports_release_38/projects/compiler-rt/make/platform/clang_darwin.mk
index b0805d9..e7fe51c 100644
--- llvm_release_38/projects/compiler-rt/make/platform/clang_darwin.mk
+++ macports_release_38/projects/compiler-rt/make/platform/clang_darwin.mk
@@ -17,8 +17,7 @@ CheckArches = \
     result=""; \
     if [ "X$(3)" != X ]; then \
       for arch in $(1); do \
-        if $(LD) -v 2>&1 | grep "configured to support" \
-             | tr ' ' '\n' | grep "^$$arch$$" >/dev/null 2>/dev/null; then \
+        if ( $(LD) -v 2>&1 | grep "configured to support" | tr ' ' '\n' | grep "^$$arch$$" >/dev/null 2>/dev/null ) || (! ( $(LD) -v 2>&1 | grep -q "configured to support" ) && test "$$arch" = "i386" -o "$$arch" = "x86_64" -o "$$arch" = "ppc" -o "$$arch" = "ppc64" ); then \
           if $(CC) -arch $$arch \
             -integrated-as \
             $(ProjSrcRoot)/make/platform/clang_darwin_test_input.c \
-- 
2.10.1

